Source/WebKit/NetworkProcess/haiku/NetworkProcessMainHaiku.cpp

```diff
@@ -46,9 +45,23 @@ class NetworkProcessMainBase: public AuxiliaryProcessMainBaseNoSingleton<Network
     }
 };
 
+class NetworProcessApp : public BApplication
+{
+public:
+    NetworProcessApp() : BApplication("application/x-vnd-HaikuWebKit-NetworkProcess") {}
+
+    void ArgvReceived(int argc, char** argv)
+    {
+        AuxiliaryProcessMain<NetworkProcessMainHaiku>(argc, argv);
+    }
+};
+
 int NetworkProcessMain(int argc, char** argv)
 {
-    return AuxiliaryProcessMain<NetworkProcessMainBase>(argc, argv);
+    NetworProcessApp* app = new NetworProcessApp();
```

Comment:
I don't understand why this code is moved outside of platformInitialize. Is that called too late to create the application in there? If so, it would be nice to have a comment here about it, so someone (probably me) doesn't try to refactor it to look more like the UNIX version.

Reply:
Alright, I gave it a whirl.

**Method 1:** Make the BApplication in the RunLoop.

The BApplication has to be constructed in the constructor to be able to queue messages. To construct it, we first need a way to figure out the app signature for the BApplication. There are two options:

1. `RuntimeApplicationChecks.h` has `isInWebProcess` and `isInNetworkProcess`. Perfect, but it's not available in WTF so we can't use it.
2. Pass the app signature to RunLoop's constructor. This isn't easy because there are many places in the code that could construct it. I believe the current method is better than trying to do this.

**Method 2:** Make the BApplication in platformInitialize.

It will be up to the run loop to actually start this. Once again, the run loop has to know whether it is running in MiniBrowser, or WebProcess/NetworkProcess to be able to know whether it should start be_app itself or not. So this won't work.

**Method 3:** Put a comment here documenting why it is the way it is.

Perfect! :slightly_smiling_face: